ccm_web/client/src/components/CreateSelectSectionWidget.tsx

```diff
@@ -28,43 +28,59 @@ const useStyles = makeStyles((theme) => ({
   }
 }))
 
-interface CreateSelectSectionWidgetProps {
+interface WidgetWithoutCreateProps {
+  readonly canCreate: false
+}
+
+interface WidgetWithCreateProps {
+  readonly canCreate: true
   course: CanvasCourseBase
+  onSectionCreated: (section: CanvasCourseSection) => void
+}
```

Comment:
Why do you need to 2 interfaces `canCreate: true/false `? Can this be something like below? I see that you are assigning `canCreate`  true or false [here](https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/298/files#diff-c30ee72622169f2c71d21d46e688fde3c6758cd5964030732fdeda72bee0aa3fR130) when calling the Create section wizard.
 I sure there must be good reason why you are doing this. But I might not catching it. 
```
interface WidgetWithoutCreateProps {
  readonly canCreate: boolean
}
```

Reply:
I can play around this some more, but this is the main way I know how to have the extra props `course` and `onSectionCreated` not be necessary when `canCreate === false`. We could just pass the props and not use them, but I leaned towards the component only enforcing their presence if they're needed.

Reply:
I think this is a standard pattern. See https://www.benmvp.com/blog/conditional-react-props-typescript/ and https://stackoverflow.com/questions/66917440/typescript-conditional-types-for-props.

Reply:
Thanks for additional info.